reporting/reporting.go

```diff
@@ -0,0 +1,127 @@
+package reporting
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/honeycombio/libhoney-go"
+
+	"github.com/honeycombio/honeytail/event"
+)
+
+const (
+	contextKeyBuilder = "reporting"
+	reportingSuffix   = "-reports"
+)
+
+func NewContext(ctx context.Context) context.Context {
+	builder := libhoney.NewBuilder()
+	return context.WithValue(ctx, contextKeyBuilder, builder)
+}
+
+func Options(ctx context.Context, options interface{}) {
+	if ev := getEvent(ctx); ev != nil {
+		optj, err := json.Marshal(options)
+		if err != nil {
+			return
+		}
+		ev.AddField("config_json", string(optj))
```

Comment:
why marshal and stringify options instead of adding them as an object and letting libhoney do it?

Reply:
Ahh, I forgot that [our `libhoney.Add`](https://github.com/honeycombio/libhoney-go/blob/master/libhoney.go#L408) respects JSON tags, I had the impression that it just used reflection.

I was mostly hoping to rely on `omitempty` to make the config file less noisy. 😶 

Ahhhhh I see, but we're not talking about `.Add()` at all, are we? And `AddField("config", theconfigobj)` does the right thing?

🙏  grateful, thanks